### PR TITLE
Itst errorhandling

### PR DESCRIPTION
**Pull request by itst**
This is the error handling only version.

I will run the scripts against more PDFs during the next few days, but I think this will already handle the worst cases. Then again, with PDF, are there any cases that are not worse :-)

### CODE REVIEW

File: pdfannots.py [hunk 1]
```diff
@@ -93,12 +93,17 @@ def __init__(self, pageno, tagname, coords=None, rect=None, contents=None):
...
-        self.rect = rect
+        if rect is not None:
+            self.rect = rect
+        else:
+            self.rect = None
         self.text = ''
 
-        if coords is None:
+
+        if coords is None or isinstance(coords, pdfminer.pdftypes.PDFObjRef):
```
[the remark this answers is not in the document]
Comment (itst) [reply pdfannots.py]: I tried resolving. I believe the PDF is off. I am still tinkering. And yes, we should check for list!

File: pdfannots.py [hunk 2]
```diff
@@ -108,6 +113,8 @@ def __init__(self, pageno, tagname, coords=None, rect=None, contents=None):
                 yvals = [y0, y1, y2, y3]
                 box = (min(xvals), min(yvals), max(xvals), max(yvals))
                 self.boxes.append(box)
+        except:
```
[the remark this answers is not in the document]
Comment (itst) [reply pdfannots.py]: True. I am trying to reproduce this error to make the error handling more meaningful ...

Do you still encounter strange PDFs in your daily work, forcing you to adopt the script?

File: pdfannots.py [hunk 3]
```diff
@@ -127,11 +134,12 @@ def gettext(self):
...
-            (x0, y0, x1, y1) = self.rect
-        elif self.boxes:
-            (x0, y0, x1, y1) = self.boxes[0]
-        else:
+        try: 
+            if self.rect:
+                (x0, y0, x1, y1) = self.rect
+            elif self.boxes:
+                (x0, y0, x1, y1) = self.boxes[0]
+        except:
```
[the remark this answers is not in the document]
Comment (itst) [reply pdfannots.py]: See above :)

I think ... We should close this MR for now. Sorry for taking so much of your time ...

File: pdfannots.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xabu) [reply pdfannots.py]: Yes, a few -- the script has acquired special cases from strange PDFs (and I've made a number of fixes to pdfminer for the same reason). OTOH 99% of my PDFs probably come from pdflatex, and I only use a single annotation app.

**Comment by 0xabu** [comment]
I _think_ https://github.com/0xabu/pdfannots/commit/e6132b6a68b2e76b258679c78dd08d3bf821648c probably fixed the issues this was intended to address.

**Pull request closed without merging**